apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/preadj/AbstractR4ResourceProvider.java

```diff
@@ -304,7 +324,16 @@ Bundle createBundleFor(
               .collect(Collectors.toList()));
     }
 
+    resources.sort(Comparator.comparing(r -> r.getIdElement().getIdPart()));
```

Comment:
Would appreciate a comment explaining the sorting strategy and how it is implemented across this call and the ordering in the query in findAllByMbiAttribute() as well as the reasons for the sort since this one is not strictly needed for the pagination functionality to be stable (I think?) but provides parity with EOB.

Just wondering as well if we should just do the same thing that EOB does and do both aspects of the sort here in one place and get rid of the orderBy on the query below?

Reply:
I added some comments to clarify the code, but to give better PR specific clarification, the FHIR resource level sorting is related to the pagination feature to ensure it mimics the EOB's sorting logic, while the entity level sorting is for a completely separate purpose of ensuring testing predictability as an additional tack on.